### PR TITLE
Numeric fixes

### PR DESCRIPTION
**Pull request by MiYanni**
Fixes: https://github.com/Azure/autorest.csharp/issues/301

Properly handles numeric formats for short, int, long, float, double, and decimal. Enables a couple more tests to pass in body-complex.

### CODE REVIEW

File: src/AutoRest.CSharp.V3/Pipeline/Extensions.cs [hunk 1]
```diff
@@ -27,9 +27,9 @@ internal static class Extensions
             AllSchemaTypes.Dictionary => null,
             AllSchemaTypes.Duration => typeof(TimeSpan),
             AllSchemaTypes.Flag => null,
-            AllSchemaTypes.Integer => typeof(int),
+            AllSchemaTypes.Integer => null, // Handled in ToFrameworkType for NumberSchema
```
Comment (pakrym) [inline src/AutoRest.CSharp.V3/Pipeline/Extensions.cs]: remove them from here?

**Comment by pakrym** [comment]
This might be a good time to start adding tests that verify model structure. In this case we can verify that property types are what we expect.

File: src/AutoRest.CSharp.V3/Pipeline/Extensions.cs (hunk 1 above)
Comment (MiYanni) [reply src/AutoRest.CSharp.V3/Pipeline/Extensions.cs]: This isn't a final design. I'm starting to get a handle of how we need to layer this stuff. Most of these mappings aren't going to hold up.

**Pull request merged**